code/lib/cli/src/generators/baseGenerator.ts

```diff
@@ -171,6 +171,15 @@ const hasInteractiveStories = (rendererId: SupportedRenderers) =>
 const hasFrameworkTemplates = (framework?: SupportedFrameworks) =>
   ['angular', 'nextjs'].includes(framework);
 
+function shouldUseSWCCompiler(builder: Builder, projectType: ProjectType) {
+  return (
+    builder === CoreBuilder.Webpack5 &&
+    projectType !== ProjectType.ANGULAR &&
+    // TODO: Remove in Storybook 8.0
+    projectType !== ProjectType.NEXTJS
+  );
+}
```

Comment:
I wonder if this should be a setting in each individual generator?

Like here:
https://github.com/storybookjs/storybook/blob/52fab185a9e30b6c69758b47264f1b56a9bd9ef8/code/lib/cli/src/generators/HTML/index.ts#L5